src/librustc_metadata/rmeta/decoder.rs

```diff
@@ -587,15 +579,24 @@ impl<'a, 'tcx> CrateMetadataRef<'a> {
         &self.raw_proc_macros.unwrap()[pos]
     }
 
-    fn item_name(&self, item_index: DefIndex) -> Symbol {
+    fn item_name(&self, item_index: DefIndex, sess: &Session) -> Ident {
```

Comment:
```suggestion
    fn item_ident(&self, item_index: DefIndex, sess: &Session) -> Ident {
```

Reply:
Depending on perf results, it may be reasonable to provide `fn get_name` as well for the cases in which the span is thrown away (only `fn get_struct_field_names` right now).

Reply:
https://github.com/rust-lang/rust/pull/70077#discussion_r394613740 still not addressed.

Reply:
I didn't realize that the suggestion was separate from your comment about perf results.